lessons/02-javascript/README.md

```diff
@@ -152,6 +157,8 @@ for (var key in object) {
 }
 ```
 
+Note that the order of iteration is not guarenteed with this syntax and later, if you use libraries which modify the prototype (more on this later) of base types, can have strange behavior.
```

Comment:
While it may be preferred. It's not for that reason:

> The Object.keys() method returns an array of a given object's own enumerable properties, in the same order as that provided by a for...in loop (the difference being that a for-in loop enumerates properties in the prototype chain as well).

Object.keys() removes inherited methods, where for...in does not (hence requiring the hasOwnProperty check)